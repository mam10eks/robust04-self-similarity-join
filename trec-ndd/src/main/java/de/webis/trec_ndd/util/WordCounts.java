package de.webis.trec_ndd.util;

import java.io.IOException;
import java.time.temporal.ChronoUnit;

import org.netspeak.client.Netspeak;
import org.netspeak.client.Request;

import com.google.common.cache.CacheBuilder;
import com.google.common.cache.CacheLoader;
import com.google.common.cache.LoadingCache;

import lombok.SneakyThrows;
import lombok.experimental.UtilityClass;
import net.jodah.failsafe.Failsafe;
import net.jodah.failsafe.RetryPolicy;

@UtilityClass
public class WordCounts {
	private static final String NETSPEAK_API = "https://api.netspeak.org/netspeak3/search?";
	
	public static RetryPolicy<Long> retryPolicy = new RetryPolicy<Long>()
			  .handle(Exception.class)
			  .withBackoff(5, 180, ChronoUnit.SECONDS)
			  .withMaxRetries(3);
	
	private static final LoadingCache<String, Long> WORD_COUNT = CacheBuilder.newBuilder()
			.build(CacheLoader.from(WordCounts::wordCountFailsave));

	@SneakyThrows
	public static Long getWordCount(String word) {
		return WORD_COUNT.get(word);
	}
	
	private static Long wordCountFailsave(String word) {
		return Failsafe.with(retryPolicy)
				.get(() -> getWordCountOrFail(word));
	}
	
	private static long getWordCountOrFail(String word) throws IOException {
		Request request = new Request();
		request.put(Request.QUERY, word);
		
		return new Netspeak(NETSPEAK_API)
				.search(request).getTotalFrequency();
	}
}
