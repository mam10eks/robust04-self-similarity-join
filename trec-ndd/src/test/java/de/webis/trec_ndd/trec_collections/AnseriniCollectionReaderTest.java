package de.webis.trec_ndd.trec_collections;

import java.util.List;

import org.approvaltests.Approvals;
import org.junit.Test;

import io.anserini.collection.SourceDocument;

public class AnseriniCollectionReaderTest<T extends SourceDocument> {
	@Test
	public void approveTransformationOfSmallCollection() {
		CollectionReader<T> reader = robustAnseriniCollectionReader();
		List<CollectionDocument> documents = reader.extractJudgedDocumentsFromCollection();

		Approvals.verifyAsJson(documents);
	}
	
	@Test
	public void approveTransformationOfSmallCluewebSampleCollection() {
		CollectionReader<T> reader = cluewebAnseriniCollectionReader();
		List<CollectionDocument> documents = reader.extractJudgedDocumentsFromCollection();

		Approvals.verifyAsJson(documents);
	}

	private static <T extends SourceDocument> CollectionReader<T> robustAnseriniCollectionReader() {
		String pathToCollection = "../collection_to_doc_vectors/test/data/robust";
		String pathToQrels = "../collection_to_doc_vectors/test/data/robust-qrels.txt";
		String collectionType = "TrecCollection";
		String documentGenerator = "JsoupGenerator";

		return new AnseriniCollectionReader<T>(pathToCollection, pathToQrels, collectionType, documentGenerator);
	}
	
	private static <T extends SourceDocument> CollectionReader<T> cluewebAnseriniCollectionReader() {
		String pathToCollection = "../collection_to_doc_vectors/test/data/clueweb09-sample";
		String pathToQrels = "../collection_to_doc_vectors/test/data/clueweb09-qrels.txt";
		String collectionType = "ClueWeb09Collection";
		String documentGenerator = "JsoupGenerator";

		return new AnseriniCollectionReader<T>(pathToCollection, pathToQrels, collectionType, documentGenerator);
	}
}
