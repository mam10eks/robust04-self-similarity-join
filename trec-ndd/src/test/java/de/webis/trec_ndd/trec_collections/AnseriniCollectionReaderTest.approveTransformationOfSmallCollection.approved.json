[
  {
    "id": "LA010189-0001",
    "content": "January 1, 1989, Sunday, Home Edition NEW FALLOUT FROM CHERNOBYL; THE SOCIAL IMPACT OF THE CHERNOBYL DISASTER BY DAVID R. MARPLES (ST. MARTIN\u0027S PRESS: $35, CLOTH; $14.95, PAPER; 316 PP., ILLUSTRATED; 0-312-02432-0) The onset of the new Gorbachev policy of glasnost, commonly mistranslated as openness but closer in connotation to candor or publicizing, has complicated the task of Soviet secret-keepers and has allowed substantial new Western insights into Soviet society. David R. Marples\u0027 new book, his second on the Chernobyl accident of April 26, 1986, is a shining example of the best type of non-Soviet analysis into topics that only recently were absolutely taboo in Moscow official circles. The author, a British-educated historian and economist, is a research associate with the Canadian Institute of Ukrainian Studies at the University of Alberta, and the academic style of the book is undisguised. However, its intended audience is the general public, and anyone interested in nuclear power, or Soviet economy and society, or human drama, or just plain sleuthing state secrets, will find hitherto unpublished revelations and explanations of the event and its continuing aftermath. The effects of Chernobyl reverberated throughout so many facets of Soviet society that a continuous coherent narrative is probably impossible. Marples discusses half a dozen major themes arranged in a fairly arbitrary order (as indicated by the frequent and helpful cross references throughout the text) and succeeds in mapping out his main themes. The personal interests of each reader determine which of the sections may be deemed too detailed and which too sketchy, but considering the need for such a comprehensive overview, the levels are generally appropriate. The book is, on the one hand, not a light read, and an executive summary might have been possible in a quarter the length. But, on the other, so many of the judgments depend on a subtle interpretation of a multitude of sources that the author is obligated to present the raw data for the reader\u0027s inspection. The modular nature of the book also allows a reader to skip, browse, and revisit earlier sections, aided by a convenient internal organization and a thorough index. First in the world\u0027s attention, and in the text, is a discussion of the human victims of the accident. The official tally is 31 (only about 20 names have ever been released), but Marples suspects there were other short-term radiation victims. A large number of unnecessary late-term abortions were also performed on local women, and by rights those unborn babies count as casualties. Widespread \"radiophobia\" led to restricted diets which created malnourishment and subsequent disease in thousands of people. The tens of thousands of people taking part in cleanup operations were never included in official totals of those exposed. Since the book went to press, Soviet military sources have referred to at least one death in the actual reactor entombment program. But the greatest toll is likely to occur with the delayed deaths. Here, Marples encounters for the first time the soon familiar theme of official Soviet myth-making around the event: Reality is twisted to serve state policy objectives, which include calming an alarmed public with assurances that all is well when it isn\u0027t. And thus is born what he properly labels the \"myth of Chernobyl,\" the official line that the disaster provided a test that Soviet society passed with honor. \"In the Soviet view,\" he writes, \"it was first and foremost a victory, a story with an ending, and an ending that was triumphant.\" Thus, when sober Western medical estimates placed the future \"excess cancer deaths\" at several tens of thousands, both in the Soviet Union and in Europe (a few tenths of a percent elevation of the natural cancer rate), the Soviets reacted furiously. The estimates are branded \"nonsense\" and the estimators are dismissed as \"panic mongers\" promulgating \"anti-Soviet venom.\" Subsequently the author addresses themes of environmental impact, economic and political repercussions, public images, and the recovery operations. Along the way, Marples provides a damning list of examples in which Soviet officials attempted to retreat behind old-style cover-ups and outright lies. False information was issued on radiation levels, on subsequent accidents at the site, on contamination levels of the Kiev water supply, on severe discipline against non-volunteer cleanup personnel, on reactor entombment schedules and on operator training levels. A severe 1986-1987 countrywide electrical power shortage was officially denied although it was real enough to compel the restart of three Chernobyl reactors in explicit violation of Soviet safety regulations. Design deficiencies of the Chernobyl-style reactors were downplayed and human errors were declared to be the primary culprit. Ultimately, observes the author, \"It is ironic that in an era of openness, Chernobyl may have been both the pioneer of glasnost under Gorbachev and then subsequently its first casualty.\" He ultimately concludes, \"Aspects of the disaster . . . have rarely been dealt with thoroughly or even honestly by Soviet sources.\" Hence the need for this book, a need which is admirably fulfilled despite the many remaining mysteries and uncertainties. The July, 1987, trial of reactor personnel marked a full circle of disclosure. Journalists were allowed into the pre-scripted first and last days, but the weeklong deliberative sessions were held in secret and no word of their substance has ever been released. The propaganda purpose of the trial and surrounding official publicity, he maintains, had one goal: \"To divert culpability from the party hierarchy, in Kiev and especially in Moscow.\" This is precisely the theme I have also encountered in my own investigations of aerospace accidents of the past. Where individual human failings led to catastrophe, a sanitized story may eventually be released, but where Kremlin policy led to disaster (such as the Nedelin catastrophe of 1960 or the Soyuz-1 disaster in 1967), the entire event remains absolutely off limits to glasnost. The closing blow-by-blow description of the nuclear power debate presages a dramatic event which occurred too recently for inclusion in this first edition. Viktor Legasov, tagged by the author as one of the country\u0027s two leading pro-nuclear advocates, actually was sinking into private despair over the poor implementation of safety standards. In the end, he made his final and most eloquent testimony to this despair on the second anniversary of the accident, by committing suicide. For several weeks the Soviets tried to sit on the circumstances of his \"tragic death,\" even issuing official non-explanations which asserted that the death was not due to medical effects of radiation. Finally, crusading journalist Vladimir Gubarev, with access to Legasov\u0027s notebooks, broke the story in Pravda. Readers of this book will come to know these and other characters so well that the suicide fits right into the \"big picture\" of the catastrophe\u0027s social impacts. For an author to so accurately describe a social milieu that subsequent unpredictable events only enhance his insights is testimony to the highest quality of scholarship. Readers of Marples\u0027 book will rarely be surprised as the Chernobyl catastrophe\u0027s consequences continue to unfold in the future."
  },
  {
    "id": "LA010189-0043",
    "content": "January 1, 1989, Sunday, Home Edition FIXING THE WRONG THING When the two national political parties begin working together to make things \"better,\" better beware. During the last campaign, Democrats and Republicans formed a commission to grab sponsorship of the last presidential debate away from the League of Women Voters. The result was that the two presidential campaigns were able to dictate ground rules that resulted not in a real debate but in an excessively restricted and programmed question-answer session. Now the parties have joined in a new bipartisan commission to do something about the national political conventions. The commission has been directed to propose ways of streamlining the conventions so that they are more appealing to Americans on television. This is in obvious response to the poor ratings that the conventions drew last summer and to network television\u0027s reluctance to carry the entire proceedings. Frank Fahrenkopf Jr., the retiring Republican national chairman, said that there had been criticism from the media that the conventions \"have become nothing more than staged activities in which nominees give canned speeches.\" The problem is that this dog is barking up the wrong tree. What needs fixing is the pre-convention nominating process that goes on forever and produces winners long before the delegates gather. The prospective nominees thus have control of the convention program and arrange things so that there is a minimum of uncertainty and controversy. This year even the names of the vice presidential running mates were known in advance, so the conventions were robbed of any remaining tidbit of suspense. What the national parties must do is shorten the primary campaign and start it someplace other than Iowa or New Hampshire. They need to be willing to have actual issues debated on the convention floor. They need some spontaneity and real action -- perhaps even a multiple roll-call contest for the nomination. They do not need more programming, but less. Fix the primary process, and the convention problem will fix itself."
  },
  {
    "id": "LA010189-0097",
    "content": "January 1, 1989, Sunday, Home Edition IN SEARCH OF AN IDENTITY; AREA LACKS A SENSE OF COMMUNITY Dick Haddon\u0027s church has an identity problem. \"I tell people that the First United Methodist Church of Gardena is not in Gardena,\" the Rev. Haddon said, \"and people just kind of stare in disbelief. \"This is Gardena, but it\u0027s not. Even the post office says \u0027Gardena,\u0027 but the reality is, we do not live in Gardena. . . . If we had a fire . . . the response would come from the city of Los Angeles.\" That is a complicated fact of life in Harbor Gateway, the 8-mile long, 4-block-wide umbilical cord acquired at the turn of the century to link Los Angeles to its port. In a city where community names conjure up images -- Venice, Silver Lake, Boyle Heights -- Harbor Gateway draws a blank. Bounded on the west by Gardena and Torrance and on the east by Carson and unincorporated county land, the area was for decades known as the \"city strip,\" the \"shoestring strip\" or, simply, \"the strip.\" Area Rechristened Four years ago, the Los Angeles City Council rechristened the area to try to bring a sense of identity and community pride to its citizens -- some of whom do not even know they live in Los Angeles. Harbor Gateway lacks much of what makes a community a community -- no central business district, no civic center or gathering place, no library branch, no police station and, as Haddon noted, no post office. Its largest park is a cemetery. And, despite the new name, mailing addresses of residents remain unchanged. They still say Torrance or Gardena, not Los Angeles. Not surprisingly, this leaves some people confused. \"They say, \u0027Look, my mailing address is Gardena,\u0027 \" reported Gene Painter, the city of Gardena\u0027s superintendent of human services who said he does not turn away Harbor Gateway residents who ask for help. \"Strange as it may seem, they don\u0027t know where they live.\" Optician Chris Toughill joined the Torrance Chamber of Commerce when he opened his shop in a Harbor Gateway strip mall, and his neighbors did the same. It wasn\u0027t until the Torrance mayor declined to attend their grand opening, Toughill said, that the merchants discovered that they were in Los Angeles. Even the young toughs of the Gardena 13 gang, interviewed while hanging out on a Harbor Gateway street corner, insisted they were standing in Gardena. In population, Harbor Gateway is tied with Westwood as Los Angeles\u0027 second-fastest growing area. Sylmar is first. Between 1980 and 1986, according to Los Angeles Planning Department estimates, Harbor Gateway\u0027s population increased 15.6%, from 30,238 to 34,951. A drive through the Gateway is like a trip from the city to the suburbs. On the north end, which begins at El Segundo Boulevard, there are apartments needing repair, gang graffiti on walls and single-family homes with barred windows. The southern end -- which stops at Sepulveda Boulevard, where Harbor City begins -- is punctuated by newly built strip malls, condominiums and well-manicured lawns. Property values follow a similar pattern, with the lowest prices in the north. Home values in nearby Torrance or Gardena, however, are considerably higher. Real estate agents report that a house in Harbor Gateway might sell for as much as $50,000 less than one across the street in another community. Agents attribute the difference to better city services, particularly in Torrance, which has its own schools. \"The minute you mention Los Angeles,\" said Gardena real estate agent John Warner, \"it frightens . . . people away. It\u0027s just one of those things. Homes have always been cheaper over there (Harbor Gateway).\" Said resident Betty Roy: \"I hate to be known to live in the strip. When my friends ask where I live, I tell them Torrance.\" Indeed, four years after the name change -- in spite of spiffy blue-and-white signs that declare \"Harbor Gateway\" below the Los Angeles city seal -- those who live and work there say they still feel betwixt and between, not fully a part of Los Angeles, Harbor Gateway or the bordering cities. \u0027In Limbo\u0027 Moreover, they complain that their new moniker has not brought with it what they really want: more attention from city officials, better services such as street cleaning and tree trimming and, above all, increased police protection. \"We\u0027re like in limbo,\" said Roy, a longtime resident of the Gateway\u0027s southern end. \"We\u0027re not really Torrance and we\u0027re not really Carson. . . . We\u0027re only here because Los Angeles needs this little piece of land to attach to the harbor. Nobody cares about us. . . . We\u0027re like the forgotten people.\" For Sylvia Figueroa, life on the border means a daily struggle at It\u0027s a Miracle, the Mexican pizza parlor she opened a year ago at Gardena Boulevard and Vermont Avenue in Los Angeles. The city of Gardena is across the street, and the differences are stark. On the Gardena side, Gardena Boulevard -- that city\u0027s main business thoroughfare -- is getting a face lift. The street is clean, and the trees are neatly trimmed. City officials are offering grants to business owners for renovations. Neighborhood Contrasts Across the street in Los Angeles, the trees are scraggly and overgrown, vagrants sleep in a littered vacant lot, liquor stores abound and the Gardena 13 gang seems to own the corner. \"The other side there\u0027s no drunk people, no people looking for drugs,\" said Figueroa, who sweeps her sidewalk regularly. \"The other side is so clean. Even the sidewalk is clean. \"You see the street?\" she asked, pointing to a pile of litter. \"That\u0027s terrible.\" Los Angeles City Councilwoman Joan Milke Flores, who represents the area, conceded that the Gateway is not getting its share of services. \"It\u0027s difficult to get a library in there, because no matter where you put a library in the Gateway, you only serve a limited number of city residents,\" she said. \"It\u0027s difficult to get a street cleaning service on a regular basis because they have to come from somewhere else to get there. . . . It\u0027s not like having a nice, round community where you have something in the middle and it serves everybody around it.\" The United Way apparently agrees. In March, the umbrella agency\u0027s Harbor/Southeast division declared Harbor Gateway an \"under-served geographic area,\" citing \"real gaps in law enforcement\" and social services. The declaration is part of a long-term plan by the United Way to upgrade Harbor Gateway by organizing neighborhood leaders to form nonprofit programs for residents. A citizens group, chaired by Haddon, is setting goals for the project, apparently marking the first time that community leaders around the Gateway have come together to talk about the area\u0027s problems. The minister and some others agree that the first goal should be improving community pride and cohesiveness to create political clout. To do that, residents must reverse 82 years of history. Over the years, since the so-called Shoestring Strip Annexation of Dec. 26, 1906, Harbor Gateway residents have somehow coalesced into five distinct neighborhoods. From north to south, their rough boundaries are El Segundo Boulevard to Rosecrans Avenue; Rosecrans to Redondo Beach Boulevard; Redondo Beach to Artesia Boulevard; Artesia to 190th Street; and 190th Street to Sepulveda Boulevard. The east-west boundaries of the three northern neighborhoods are Vermont Avenue and Figueroa Street, while Western and Normandie avenues bound the two in the south. Overall, Harbor Gateway is a racially mixed, middle-income area, with an estimated median household income of $30,431 and a population that is 48.4% white, 13.2% black, 16.5% Asian and 21.8% other racial groups, according to the 1980 census, the most recent statistics available. Latinos -- whose numbers are not tallied as part of the racial breakdown in the 1980 census -- are Harbor Gateway\u0027s largest minority group, accounting for 32.9% of the overall population, according to census data. However, the picture changes when the area is more closely examined. Demographic statistics show that the various Gateway neighborhoods often resemble adjacent communities more closely than they do one another. For example, the two Gateway neighborhoods near south South-Central Los Angeles and Compton have the heaviest concentration of blacks, 44.7%. Whites make up 23.6% of those two neighborhoods, which extend north from Redondo Beach Boulevard, and Asians make up 6.9%. The estimated median household income there is $27,453. Make-Up of Community But in the strip\u0027s midsection, from Artesia Boulevard to 190th Street, the population reflects that of the southern end of Gardena, which is heavily Asian. There, the black population is minuscule -- just 0.4% -- with most of the population split between whites, 48.1%, and Asians, 44.8%. Estimated household income in this section jumps dramatically, to $44,498. And south of 190th Street, approaching predominantly white Torrance, the white population jumps even higher, to 66.3%, while the Asian population dips to 8.6% and the number of blacks rises slightly, to 2%. Median household income is $27,607. Although it is primarily residential, the Gateway also includes a burgeoning business district at the San Diego and Harbor freeways, where, according to the California Department of Transportation, 500,000 cars drive by each day, making it the second-busiest intersection in Southern California. The so-called 190th Street Corridor is a major drawing card for commercial developers, who have built about 5 million square feet of office space there in the last eight years. Gleaming high-rises with pleasant landscaping have replaced a Shell oil refinery and manufacturing plants. \"We sort of had a shift from the Industrial Age to the Information Age,\" said Howard Mann, president of Andrex Development Co., one of the area\u0027s major developers. Mann predicts that the move away from industrial to commercial uses will help upgrade nearby areas by attracting home buyers who hold high-paying jobs in what he calls Harbor Gateway\u0027s \"urban cluster.\" Already, he is planning a restaurant to meet that market, based on an analysis that, he said, \"found that the average income per household within a 5-mile radius of this intersection is $50,000.\" Dining in elegance, however, is certainly not a priority farther north in Harbor Gateway, where large concentrations of undocumented Latinos residents are struggling to get by. A study by the United Way found that illegal Latino immigrants are the \"most under-served\" group in the Gateway. Haddon and other local leaders said their presence also makes it difficult to form neighborhood coalitions because illegal aliens fear they will attract unwanted attention from immigration authorities if they participate in community affairs. \"Their need is to be invisible,\" the minister said, \"and that does not make for a good community.\" Haddon is particularly attuned to the problems of undocumented residents because his church runs the 3-year-old Harbor Gateway Center, one of only two social service agencies the United Way found serving the strip. Center\u0027s Problems The center -- which offers a variety of nutritional, health and educational programs to about 1,500 people -- lacks money and staff. The executive director left more than a year ago when the money to pay her ran out. Haddon, who took the pastor\u0027s job in July, has been doing her job ever since. United Way officials say they would like to strengthen the center and also persuade social service agencies in neighboring communities to expand operations here. Their study showed that the area lacks adequate shelters for battered women, delinquency prevention programs, drug- and alcohol-abuse programs and crisis intervention centers. If there is one common thread in the complaints from Gateway residents, it is the increasing gang activity and the lack of police protection. Many said that they go for days without seeing a patrol car and that officers covering the area are often sent on calls elsewhere. Los Angeles police officials said they have been trying to improve service, especially during the past year when citizens groups and Flores have pressed for additional patrols. But they also acknowledged that cars are called out of the area more often than they would like. Capt. Sandy Wasson of the Los Angeles Police Department\u0027s Southeast Division, which covers the part of the strip north of the San Diego Freeway, took issue with residents who say Harbor Gateway gets less than its fair share of coverage. He said a recent deployment study showed that while the area has fewer patrol cars -- just one basic car in the north and one in the south -- it also has less crime. Wasson said he views Harbor Gateway as \"a neighborhood in transition that\u0027s got great potential.\" He has encouraged the formation of neighborhood watch groups. Flores, for one, said she senses a growing participation in such groups and \"a renewed interest in community improvement.\" She maintained that the city has improved services in recent years, especially police coverage. She also noted that a Harbor Gateway paramedic unit was established in June and that this year, the city Library Department finally included a proposal for a Harbor Gateway branch in its long-term plan. The councilwoman said she would like the Gateway to have its own post office and ZIP code. \"I think it\u0027s important,\" she said, \"to have an actual tag and title on who you are and where you are.\" But she said she knows persuading the federal government will be difficult. In the meantime, Flores said, she is thankful for whatever recognition the new name has brought the area. \"At least,\" she said, \"I know that when I mention Harbor Gateway in City Council now, my colleagues know what I\u0027m talking about.\""
  },
  {
    "id": "LA010189-0152",
    "content": "January 1, 1989, Sunday, Bulldog Edition"
  },
  {
    "id": "LA010189-0192",
    "content": "January 1, 1989, Sunday, Orange County Edition STOCK WATCH: SOME COMPANIES HIT HARDEST BY CRASH PICKED UP THE PIECES, CAME OUT AHEAD On Friday, the last trading day of 1988, the Dow Jones Industrial Average briefly climbed above the level it had enjoyed before the October, 1987, market collapse. It was a fitting end to a year that proved stronger than expected, with the Dow increasing 12%. It was equally fitting that the best performing Orange County stocks of 1988 were among those hardest hit by the crash. Orange County\u0027s star performers managed to rescue themselves from the rubble to either approach or surpass their pre-crash highs. In most cases, analysts said, they are solid companies that enjoyed healthy returns. ICN Biomedicals, a Costa Mesa medical instruments and chemicals company, led the pack on a percentage basis. The stock rose 331%, starting the year at $3.25 and finishing at $14. For the first 9 months of its current fiscal year, ICN Biomedicals has enjoyed a 35% increase in earnings. Its third-quarter performance was even stronger, with profits up 59%. Gradco Corp., an Irvine manufacturer of office copier and printer products, placed second, appreciating 126% for the year. Gradco stock, which began the year at $5.875, closed at $13.25. In the fiscal year ended March, 1988, Gradco reported net income of $4.1 million on sales of $90.9 million, and the company continued to post strong numbers over the latter part of the year. During the past 12 months, most investors steered clear of small companies. Many had been burned badly by the crash. \"A lot of the smaller Orange County companies are not very liquid, and they just got hammered in the crash, as people fell over themselves to get out of them,\" said Larry Selwitz, director of research at Cruttenden \u0026 Co., a Newport Beach brokerage. Many investors who got back into the market in 1988 initially chose to buy stocks in larger, more established companies. But as the year wore on and many Orange County companies began to report strong earnings, investors started funneling money into such companies as ICN Biomedicals, Ropak and Gradco. \"The stock of some of these companies had become so oversold that people realized these prices offered unusually good value,\" Selwitz said. \"It was an opportunity for people to make a lot of money.\" Meanwhile, the county\u0027s biggest market losers generally reflected unspectacular corporate performance. Resdel Industries, a Newport Beach electronics firm, lost the most ground in 1988. Its stock began the year at $5.25 a share and crashed to $1.4375, a 73% decline. The company posted an $8.3-million loss for the year ended June 30 and disclosed in October that it had laid off 130 people, or more than a third of its work force. The firm\u0027s problems were attributed primarily to unsuccessful efforts to expand through acquisitions. Another bomb was Diceon Electronics, which began the year trading at $30.50 but declined to $12.50 at year end, losing 59% of its value. Diceon had sales of $147 million and earned $9.8 million in the fiscal year ended in September. But most of its profits were realized in the first 6 months of the year, while the second half was a dud. Analysts are projecting that the company will have flat earnings for the first half of 1989. As a result, Diceon, which for several years has been one of Orange County\u0027s top performing growth companies, was snubbed by analysts and investors. Preston Athey, a growth fund manager for T. Rowe Price in Baltimore, said he continues to hold Diceon and several other Orange County stocks because of what he believes are bright prospects for small growth companies in 1989. T. Rowe Price has about $1.3 billion invested in emerging growth funds. The investment firm\u0027s Orange County holdings include National Education of Irvine, Silicon Systems of Tustin and Archive Corp. Athey said a statistical indicator used by T. Rowe Price recently signaled that many, small growth companies \"are grossly undervalued\" relative to historic norms. \"If the past is any indication, 1989 is the year that these stocks will do very well relative to the rest of the market,\" Athey said. He said he is particularly bullish about small growth companies because many economists and analysts are projecting that their earnings growth will outpace larger companies in 1989. WINNERS AND LOSERS AMONG COUNTY STOCKS The top 10 percentage gainers and losers in 1988 among stocks of Orange County companies trading at $3 or more a share. Greatest Percentage Gains % Greatest Percentage Losses % 1.ICN Biomedicals +331 1.Resdel Industries -73 2.Gradco Systems +126 2.American Pacesetter -82 3.Ropak Corp. +122 3.Diceon Electronics -59 4.Nichols Institute +113 4.Silvercrest -45 5.Trimedyne +100 5.MAI Basic 4 -40 6.Laser Precision +100 6.US Facilities -36 7.Newport Electronics + 97 7.National Lumber -35 8.FHP +90 8.Computer Automation -34 9.J M Peters +86 9.Filenet -33 10.Micro D +77 10.EIP Microwave -31 Source: Newport Securities Corp."
  }
]